xbmc/filesystem/ZipFile.cpp

```diff
@@ -399,12 +399,42 @@ int CZipFile::UnpackFromMemory(std::string& strDest, const std::string& strInput
     if (!isGZ)
     {
       CZipManager::readHeader(strInput.data()+iPos,mZipItem);
-      if( mZipItem.header != ZIP_LOCAL_HEADER )
+      if (mZipItem.header == ZIP_DATA_RECORD_HEADER)
+      {
+        // this header concerns a file we already processed, so we can just skip it
+        iPos += DREC_SIZE;
+        continue;
+      }
+      if (mZipItem.header != ZIP_LOCAL_HEADER)
         return iResult;
       if( (mZipItem.flags & 8) == 8 )
```

Comment:
Are you sure you need the == ?

Reply:
No, of course not :smile: I left it as it was from the original author.